### ISSUE TITLE
new rule - error when `@property` getter takes a parameter with a default value

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#4522]
```py
class Foo:
    @property
    def value(self, huh: str = "") -> int: # should be an error
        return 1
```
https://play.ty.dev/afa9489a-afe6-4d5e-b513-30f9870a6602

i initially suspected this could be fixed in typeshed with https://github.com/python/typeshed/issues/4731, but since this parameter has a default value, the function is considered a subtype of `Callable[[Any], None]`, so i think it may need to be special-cased

**Comment by AlexWaygood** [comment]
Hmm, this one doesn't really feel like a typing error (it looks like a mistake, but it's never going to lead to an exception when the property is accessed at runtime), nor does it seem like an error that you'd really need sophisticated type inference to be able to detect reliably. To me this feels like it should be flagged by a linter like ruff rather than a type checker

**Comment by MichaReiser** [comment]
Ruff already supports this rule https://docs.astral.sh/ruff/rules/property-with-parameters/

**Comment by DetachHead** [comment]
oops, i have that rule enabled in my project so not sure why i didn't notice that, thanks
